wandb/integration/yolov8/utils.py

```diff
@@ -0,0 +1,173 @@
+"""
+Utilities for parsing and plotting Yolo results using wandb.
+"""
+import pathlib
+from copy import deepcopy
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import cv2
+import numpy as np
+import torch
+from ultralytics.yolo.utils.ops import xywh2xyxy
+
+import wandb
+from wandb import Image
+
+
+def convert_to_wb_images(
+    batch: Dict[str, Union[torch.Tensor, np.ndarray]],
+    cls: Union[torch.Tensor, np.ndarray],
+    bboxes: Union[torch.Tensor, np.ndarray],
+    masks: Optional[Union[torch.Tensor, np.ndarray]],
+    batch_idx: Union[torch.Tensor, np.ndarray],
+    names: Dict[int, str],
+) -> Tuple[List[Tuple[Image, Image]], List[Dict[Any, Union[float, Any]]]]:
+    """Utility to convert a batch of images, bounding boxes and masks from YOLO validator to list of wandb.Image objects
+    Additionally, returns a list of dictionaries with the average confidences for each class
+    """
+    images = batch["img"]
+    captions = batch["im_file"]
```

Comment:
couldn't find reference to `batch["im_file"]` in the yolov8 repo

Reply:
maybe keep the original meaning (based on name) and only name it as caption when you actually transform it?

Reply:
changed this to `image_paths` instead.